libraries/rush-lib/src/cli/actions/DeployAction.ts

```diff
@@ -101,6 +103,21 @@ export class DeployAction extends BaseRushAction {
     });
   }
 
+  private _getDependencyProjects(project: RushConfigurationProject): Set<RushConfigurationProject> {
+    const projects: Set<RushConfigurationProject> = new Set();
+    const queue: RushConfigurationProject[] = [project];
+
+    while (queue.length > 0) {
+      const _project: RushConfigurationProject = queue.shift()!;
+
+      for (const dependency of _project.dependencyProjects) {
+        queue.push(dependency);
```

Comment:
I wonder visited projects should be ignored.

To illustrate:

```
const visited: Set<RushConfigurationProject> = new Set();

...

for (const dependency of _project.dependencyProjects) {
  if (visited.has(dependency) {
    continue;
  }
  
  ...

  visited.add(dependency);
}

Reply:
Yes, it would have some performance improvement.